### PR TITLE
removed panics from AddFileToZip and UnZipSingleFile

### PR DESCRIPTION
**Pull request by kuldeep0020**
This PR is just for the cleanup of panics in misc.go

## Type of change
- [x] Cleanup
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added unit tests for the code
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: utils/misc/misc.go [hunk 1]
```diff
@@ -274,30 +274,42 @@ func AddFileToZip(zipWriter *zip.Writer, filename string) error {
...
 	if err != nil {
-		panic(err)
+		return err
 	}
-	defer r.Close()
-	inputfile := r.File[0]
+	defer func(r *zip.ReadCloser) {
+		err := r.Close()
+		if err != nil {
+			pkgLogger.Errorf("error closing the ReadCloser")
```
Comment (lvrach) [inline utils/misc/misc.go:292]: Shall we mention the error as well
```suggestion
			pkgLogger.Errorf("error closing the ReadCloser: %s", err)
```

File: utils/misc/misc.go [hunk 2]
```diff
@@ -274,30 +274,42 @@ func AddFileToZip(zipWriter *zip.Writer, filename string) error {
...
+	err = os.MkdirAll(filepath.Dir(outputFile), os.ModePerm)
+	if err != nil {
+		return err
+	}
+	outFile, err := os.OpenFile(outputFile, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, inputFile.Mode())
 	if err != nil {
-		panic(err)
+		return err
+	}
+	rc, _ := inputFile.Open()
```
Comment (lvrach) [inline utils/misc/misc.go:305]: err is being ignored here

File: utils/misc/misc.go [hunk 3]
```diff
@@ -274,30 +274,42 @@ func AddFileToZip(zipWriter *zip.Writer, filename string) error {
...
-		panic(err)
+		return err
 	}
 	_, err = io.Copy(writer, fileToZip)
 	return err
 }
 
 // UnZipSingleFile unzips zip containing single file into ouputfile path passed
-func UnZipSingleFile(outputfile string, filename string) {
+func UnZipSingleFile(outputFile string, filename string) error {
```
Comment (lvrach) [inline utils/misc/misc.go:284]: If this function is not used I would propose to delete it. It doesn't have any tests and they are things to be fixed anyway.

File: utils/misc/misc.go (hunk 1 above)
Comment (kuldeep0020) [reply utils/misc/misc.go:292]: removed the method

File: utils/misc/misc.go (hunk 2 above)
Comment (kuldeep0020) [reply utils/misc/misc.go:305]: removed the method

File: utils/misc/misc.go (hunk 3 above)
Comment (kuldeep0020) [reply utils/misc/misc.go:284]: removed

File: utils/misc/misc.go [hunk 4]
```diff
@@ -255,12 +255,12 @@ func AddFileToZip(zipWriter *zip.Writer, filename string) error {
 	// Get the file information
 	info, err := fileToZip.Stat()
 	if err != nil {
-		panic(err)
+		return err
```
Comment (lvrach) [inline utils/misc/misc.go:258]:
```suggestion
		return fmt.Errorf("file stat: %w", err)
```

File: utils/misc/misc.go [hunk 5]
```diff
@@ -255,12 +255,12 @@ func AddFileToZip(zipWriter *zip.Writer, filename string) error {
...
 	}
 
 	header, err := zip.FileInfoHeader(info)
 	if err != nil {
-		panic(err)
+		return err
```
Comment (lvrach) [inline utils/misc/misc.go:263]: return fmt.Errorf("zip header: %w", err)

File: utils/misc/misc.go [hunk 6]
```diff
@@ -274,31 +274,12 @@ func AddFileToZip(zipWriter *zip.Writer, filename string) error {
 
 	writer, err := zipWriter.CreateHeader(header)
 	if err != nil {
-		panic(err)
+		return err
```
Comment (lvrach) [inline utils/misc/misc.go:277]: return fmt.Errorf("create header: %w", err)

**Review by lvrach — APPROVED** [review]

**Review by cisse21 — APPROVED** [review]

**Pull request merged**